### ISSUE TITLE
BUG: Charge structure not parsed

### ISSUE THREAD

**Issue by jmmshn** [materialsproject/atomate2#154]
When parsing the following directory into a TaskDocument.
The result should have +1 charge (NELECT = 349 vs 350 for the neutral)
The current version of the Vasprun Object in pymatgen gives `Vasprun.structure.charge == -1` due to a bug fix in 
https://github.com/materialsproject/pymatgen/pull/2577

Either way, with or without the fix to the sign,the current parsing of the directory using `TaskDocument` gives zero charge: 
```
tdoc = TaskDocument.from_directory("./")
```
Gives:
```
tdoc.structure.charge # --> 0
```

Files to reproduce linked below:

[charged_vasprun_ex.zip](https://github.com/materialsproject/atomate2/files/9069337/charged_vasprun_ex.zip)

**Comment by jmmshn** [comment]
So it looks like this is cause:

https://github.com/materialsproject/atomate2/blob/604786224c7f621c2c74fb177d3a240f9096f405/src/atomate2/vasp/schemas/calculation.py#L487

We use `final_structure`:
https://github.com/materialsproject/pymatgen/blob/d10c460b2b20b8802e345491bfe2301d1376650e/pymatgen/io/vasp/outputs.py#L496

Which calls `_parse_structure` that reconstructs the structure from the XML data without charge.

**Comment by jmmshn** [comment]
resolved in #155
